hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -91,7 +90,7 @@ object DataSourceReadOptions {
 
   val ENABLE_HOODIE_FILE_INDEX: ConfigProperty[Boolean] = ConfigProperty
     .key("hoodie.file.index.enable")
-    .defaultValue(true)
+    .defaultValue(false)
```

Comment:
This is to run the CI on `hoodie.file.index.enable=false` to ensure the relation functions without the `HoodieFileIndex`.  This line of change is going to be reverted before merging.

Reply:
@yihua let's make sure we at least run TestCOW*, TestMOR* tables w/ this config off as well

Reply:
Don't forget to revert this

Reply:
Reverted.